src/prefect/cli/describe.py

```diff
@@ -0,0 +1,225 @@
+import click
+import pendulum
+from tabulate import tabulate
+
+from prefect.client import Client
+from prefect.utilities.graphql import with_args, EnumValue
+
+
+@click.group(hidden=True)
+def describe():
+    """
+    Describe commands that render JSON output of Prefect object metadata.
+
+    \b
+    Usage:
+        $ prefect describe [OBJECT]
+
+    \b
+    Arguments:
+        flow-runs   Describe flow runs
+        flows       Describe flows
+        tasks       Describe tasks
+
+    \b
+    Examples:
+        $ prefect describe flows --name My-Flow --version 2
+        {
+            "name": "My-Flow",
+            "version": 2,
+            "project": {
+                "name": "Test-Project"
+            },
+            "created": "2019-05-08T23:04:58.984132+00:00",
+            "description": null,
+            "parameters": [],
+            "archived": false,
+            "storage": {
+                "type": "Docker",
+                "flows": {
+                    "My-Flow": "/root/.prefect/My-Flow.prefect"
+                },
+                "image_tag": "944444e8-8862-4d04-9e36-b81ab15dcaf6",
+                "image_name": "z4f0bb62-8cc1-49d9-bda3-6rf53b865ea5",
+                "__version__": "0.5.3",
+                "registry_url": "myregistry.io/flows/"
+            },
+            "environment": {
+                "type": "CloudEnvironment",
+                "__version__": "0.5.3"
+            }
+        }
+    """
+    pass
+
+
+@describe.command(hidden=True)
+@click.option("--name", "-n", required=True, help="A flow name to query.", hidden=True)
+@click.option("--version", "-v", type=int, help="A flow version to query.", hidden=True)
+@click.option("--project", "-p", help="The name of a project to query.", hidden=True)
+def flows(name, version, project):
+    """
+    Describe a Prefect flow.
+
+    \b
+    Options:
+        --name, -n      TEXT    A flow name to query                [required]
+        --version, -v   INTEGER A flow version to query
+        --project, -p   TEXT    The name of a project to query
+    """
+    query = {
+        "query": {
+            with_args(
+                "flow",
+                {
+                    "where": {
+                        "_and": {
+                            "name": {"_eq": name},
+                            "version": {"_eq": version},
```

Comment:
What happens if I don't provide version or project? Will GraphQL ignore this equality?

Reply:
Yeah it places in `null` and that aspect in the there clause is effectively ignored

Reply:
Awesome